src/lambda/events/generate_test_event.py

```diff
@@ -42,81 +45,111 @@
-    description='Generate a json file of a mocked SNS event for testing.')
+    description='Generate a json file of a mocked SQS event for testing.')
   parser.add_argument('--synapse-project-id',
     default='syn26721259',
     help='Synapse ID of the BridgeDownstreamTest project')
+  parser.add_argument('--dataset-name',
+    default='bridge-downstream-test-dataset',
+    help=('Optional. Name of the Synapse dataset containing test data. '
+          'Default is "bridge-downstream-test-dataset".'))
   args = parser.parse_args()
   return args
 
 
-def main():
-  args = read_args()
-  project_id = args.synapse_project_id
-
+def get_synapse_client(ssm_parameter):
+  '''Get an instance of the synapse client'''
+  ssm_client = boto3.client('ssm')
+  token = ssm_client.get_parameter(
+    Name=ssm_parameter,
+    WithDecryption=True)
   syn = synapseclient.Synapse()
-  syn.login()
-
-  folder_name = 'test-data'
-  print(f'Get folder id')
-  response = list(syn.getChildren(project_id, includeTypes=['folder']))
-  folder = next(item for item in response if item['name'] == folder_name)
-  folder_id = '' if folder is None else folder.get('id')
-  if not folder_id:
-    print(f'No folder {folder_name} exists')
+  synapse_auth_token = token['Parameter']['Value']
+  syn.login(authToken=synapse_auth_token, silent=True)
+  return syn
+
+
+def get_dataset_id_by_name(syn, parent_id, dataset_name):
+  response = list(syn.getChildren(parent_id, includeTypes=['dataset']))
+  dataset = next(item for item in response if item['name'] == dataset_name)
+  dataset_id = '' if dataset is None else dataset.get('id')
+  if not dataset_id:
+    print(f'No dataset {dataset_name} exists')
     sys.exit(1)
+  return dataset_id
+  return dataset
+
+
+def get_latest_stable_dataset(syn, dataset_metadata):
+  '''Gets the most recently snapshotted dataset
 
-  message_template = create_message_template(project_id, folder_id)
-  print(f'Fetching children of synapse id {folder_id}...')
-  response = list(syn.getChildren(folder_id, includeTypes=['file']))
+  A snapshotted dataset is a 'stable' version,
+  in contrast with the current latest version of the dataset,
+  which is the 'draft' version. The most recently snapshotted dataset
+  is therefore the version one less than the current latest version.
+  '''
+  if dataset_metadata.get("versionNumber") < 2:
+    raise Exception("No stable dataset version found.")
+  stable_dataset_version = int(dataset_metadata["versionNumber"]) - 1
+  stable_dataset_id = f"{dataset_metadata['id']}.{stable_dataset_version}"
+  stable_dataset = syn.tableQuery(f"SELECT * FROM {stable_dataset_id}").asDataFrame()
+  return stable_dataset
+
+
+def main():
+  args = read_args()
+  syn = get_synapse_client(ssm_parameter=SSM_PARAMETER)
+  test_dataset_id = get_dataset_id_by_name(
+          syn=syn,
+          parent_id=args.synapse_project_id,
+          dataset_name=args.dataset_name)
+  test_dataset_metadata = syn.restGET(f"/entity/{test_dataset_id}")
```

Comment:
Nit: Can you add a comment here saying Datasets not supported which is why its a `restGET`

Reply:
Sure.